tests/run-make/output-type-permutations/rmake.rs

```diff
@@ -0,0 +1,541 @@
+// In 2014, rustc's output flags were reworked to be a lot more modular.
+// This test uses these output flags in an expansive variety of combinations
+// and syntax styles, checking that compilation is successful and that output
+// files are exactly what is expected, no more, no less.
+// See https://github.com/rust-lang/rust/pull/12020
+
+use run_make_support::{
+    bin_name, dynamic_lib_name, filename_not_in_denylist, fs_wrapper, rust_lib_name, rustc,
+    shallow_find_files, static_lib_name,
+};
+
+// Each test takes 4 arguments:
+// `must_exist`: output files which must be found - if any are absent, the test fails
+// `can_exist`: optional output files which will not trigger a failure
+// `dir`: the name of the directory where the test happens
+// `rustc_invocation`: the rustc command being tested
+// Any unexpected output files not listed in `must_exist` or `can_exist` will cause a failure.
+fn assert_expected_output_files(expectations: Expectations, rustc_invocation: impl Fn()) {
+    let must_exist = expectations.expected_files;
+    let can_exist = expectations.allowed_files;
+    let dir = expectations.test_dir;
+
+    fs_wrapper::create_dir(&dir);
+    rustc_invocation();
+    for file in must_exist {
+        fs_wrapper::remove_file(dir.to_owned() + "/" + &file);
```

Comment:
`test_dir` could be `PathBuf`, when you're anyway treating it as a path here. Who knows how `/` will behave on Windows (although it probably should handle it).

Reply:
> Who knows how / will behave on Windows

It should have no effect, since `fs_wrapper::remove_file` takes an `AsRef<Path>`.

Reply:
That doesn't automatically make it work, e.g. on Linux with backwards slashes it wouldn't work: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=1eb613878a6de3ba1eceac811eca2211

But hopefully Windows will be able to deal with forward slashes :)

Reply:
That's interesting. Well, let's see the output of the msvc try-job, it should work this time. I hope.

Reply:
> `test_dir` could be `PathBuf`, when you're anyway treating it as a path here. Who knows how `/` will behave on Windows (although it probably should handle it).

Yeah please pass a PathBuf here and use join which is *designed* to properly handle platform separators, please don't use string concat here with `/`.